File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int bp = 0;
        int n = nums.size();
        for (int i = 0; i < n - 1; i++) {
            if (nums[i] > nums[i+1]) { 
                bp++;
            }
        }
        if (nums[n - 1] > nums[0]) {
            bp++;
        }
        return bp <= 1;
    }
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int b=0;
        for (int i=0;i<nums.length-1;i++){
            if(nums[i]>nums[i+1]){
                b++;
                if (b==2){return false;}
            }
        }
        if(nums[0]>=nums[nums.length-1] || b==0){return true;}
        else{return false;}
    }
}